### DISCUSSION TITLE
Format.EdgeListTabOne does not work for weighted graph

### DISCUSSION THREAD

**Discussion by yushangdi** [networkit/networkit#1090] [Q&A]
For example, G = nk.readGraph("../input/alphabet.edgelist", nk.Format.EdgeListTabOne) gives the following error:

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File networkit/graphio.pyx:1128, in networkit.graphio.readGraph()

File networkit/graphio.pyx:79, in networkit.graphio.GraphReader.read()

RuntimeError: Scanning node failed. The file may be corrupt.

During handling of the above exception, another exception occurred:

OSError                                   Traceback (most recent call last)
Input In [7], in <cell line: 1>()
----> 1 G = nk.readGraph("../input/alphabet.edgelist", nk.Format.EdgeListTabOne)

File networkit/graphio.pyx:1126, in networkit.graphio.readGraph()

File networkit/graphio.pyx:1131, in networkit.graphio.readGraph()

OSError: ../input/alphabet.edgelist is not a valid Format.EdgeListTabOne file: Scanning node failed. The file may be corrupt.

**Comment by fabratu** [comment]
Hi, thanks for posting this issue!

The documentation is not really clear for this specific input. The `alphabet.edgelist`[1] file contains non-numerical node ids (`a`, `b`, ...). Because of this, you have to use the edgelist file reader with `continuous` set to `False`. 

Here is a working code-snippet:

```
>>> import networkit as nk
>>> reader = nk.graphio.EdgeListReader('\t', 0, commentPrefix='#', continuous=False, directed=False)
>>> G = reader.read('input/alphabet.edgelist')
>>> print(G)
NetworKit.Graph(n=5, m=4)
>>> G.weight(0,1)
3.0
```

You can also pass this parameter to `nk.readGraph` directly:

```
G = nk.readGraph('input/alphabet.edgelist',nk.Format.EdgeList, continuous=False, firstNode=0, separator='\t')
```

Also see here for more details: https://networkit.github.io/dev-docs/python_api/graphio.html?highlight=edgelistreader#networkit.graphio.EdgeListReader

[1] https://github.com/networkit/networkit/blob/master/input/alphabet.edgelist
